modules/contact/test/tests/mortar_dynamics/block-dynamics-reference.i

```diff
@@ -0,0 +1,311 @@
+    gamma = 0.5
+    execute_on = 'timestep_end'
+  []
+  [accel_y]
+    type = NewmarkAccelAux
+    variable = accel_y
+    displacement = disp_y
+    velocity = vel_y
+    beta = 0.25
+    execute_on = 'timestep_end'
+  []
+  [vel_y]
+    type = NewmarkVelAux
+    variable = vel_y
+    acceleration = accel_y
+    gamma = 0.5
+    execute_on = 'timestep_end'
+  []
+  [kinetic_energy]
+    type = KineticEnergyAux
+    block = '1 2'
+    variable = kinetic_energy
+    newmark_velocity_x = vel_x
+    newmark_velocity_y = vel_y
+    newmark_velocity_z = 0.0
+    density = density
+  []
+  [elastic_energy]
+    type = ElasticEnergyAux
+    variable = elastic_energy
+    block = '1 2'
+  []
+[]
+
+[Constraints]
+# Not using 'dynamic' constraints results in poor enforcement of contact
+# constraints and lack of kinetic and elastic energy conservation.
+  [weighted_gap_lm]
+    type = ComputeDynamicWeightedGapLMMechanicalContact
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = normal_lm
+    disp_x = disp_x
+    disp_y = disp_y
+    newmark_beta = 0.25
+    newmark_gamma = 0.5
+    use_displaced_mesh = true
+    interpolate_normals = false
+    # Capture tolerance is important. If too small, stabilization takes longer
+    capture_tolerance = 1.0e-5
+    c = 1.0e6
+  []
+
+  [normal_x]
+    type = NormalMortarMechanicalContact
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = normal_lm
+    secondary_variable = disp_x
+    component = x
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+    interpolate_normals = false
+  []
+  [normal_y]
+    type = NormalMortarMechanicalContact
+    primary_boundary = 20
+    secondary_boundary = 10
+    primary_subdomain = 4
+    secondary_subdomain = 3
+    variable = normal_lm
+    secondary_variable = disp_y
+    component = y
+    use_displaced_mesh = true
+    compute_lm_residuals = false
+    interpolate_normals = false
+  []
+[]
+
+[BCs]
+  [botx]
+    type = DirichletBC
+    variable = disp_x
+    boundary = 40
+    value = 0.0
+  []
+  [boty]
+    type = DirichletBC
+    variable = disp_y
+    boundary = 40
+    value = 0.0
+  []
+  [topy]
+    type = FunctionDirichletBC
+    variable = disp_y
+    boundary = 30
+    function = '${starting_point} * cos(2 * pi / 4 * t) + ${offset}'
+  []
+  [leftx]
+    type = FunctionDirichletBC
+    variable = disp_x
+    boundary = 30 # 50
+    function = '0' # '1e-2*t'
+  []
+[]
+
+[Executioner]
+  type = Transient
+  end_time = 0.275 # 8.0
+  dt = 0.025
+  dtmin = .025
+  solve_type = 'NEWTON'
+  petsc_options = '-snes_converged_reason -ksp_converged_reason -pc_svd_monitor '
+                  '-snes_linesearch_monitor'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_shift_amount'
+  petsc_options_value = 'lu       NONZERO               1e-15'
+  l_max_its = 100
```

Comment:
Hopefully you take far less than 100 linear iterations for LU and Newton. Hopefully 2 max?

Reply:
All these tests run fine with Newton, which is typically the case for many applications. I recently switched the solution strategy to Newton from PJFNK and didn't take the time to go over all the settings. That line has been removed.